Simplified/Accounts/AgeCheck/NYPLAgeCheck.swift

```diff
@@ -0,0 +1,121 @@
+import Foundation
+
+protocol NYPLAgeCheckValidationDelegate: class {
+  var minYear : Int { get }
+  var currentYear : Int { get }
+  var birthYearList : [Int] { get }
+  var ageCheckCompleted : Bool { get set }
+  
+  func isValid(birthYear: Int) -> Bool
+  
+  func ageCheckCompleted(_ birthYear: Int)
+  func ageCheckFailed()
+}
+
+@objc protocol NYPLAgeCheckVerifying {
+  func verifyCurrentAccountAgeRequirement(userAccountProvider: NYPLUserAccountProvider,
+                                          currentLibraryAccountProvider: NYPLCurrentLibraryAccountProvider,
+                                          completion: ((Bool) -> ())?) -> Void
+}
+
+@objc protocol NYPLAgeCheckChoiceStorage {
+  var userPresentedAgeCheck: Bool { get set }
+}
+
+@objcMembers final class NYPLAgeCheck : NSObject, NYPLAgeCheckValidationDelegate, NYPLAgeCheckVerifying {
+  
+  // Members
+  let serialQueue = DispatchQueue(label: "\(Bundle.main.bundleIdentifier!).ageCheck")
+  var handlerList = [((Bool) -> ())]()
+  var isPresenting = false
+  var ageCheckCompleted: Bool = false
+  let ageCheckChoiceStorage: NYPLAgeCheckChoiceStorage
+  
+  var minYear: Int = 1900
+  
+  var currentYear: Int = Calendar.current.component(.year, from: Date())
+  
+  var birthYearList: [Int] {
+    return Array(minYear...currentYear)
```

Comment:
With this, the user can use 2021 as year of birth. Is this intended?

Reply:
There isn't a requirement on the upper bound of birth year, so I just use the current year.